.github/workflows/test-ghaf-infra.yml

```diff
@@ -1,21 +1,110 @@
 # SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
 # SPDX-License-Identifier: Apache-2.0
 
-name: Test ghaf-infra
+name: Run pre-push checks
 
 on:
   push:
     branches:
       - main
-  pull_request:
+  pull_request_target:
     branches:
       - main
 
 jobs:
+  # Checks if the author of pull request is in our predefined list of authorized users
+  check-identity:
+    runs-on: ubuntu-latest
+    outputs:
+      authorized_user: ${{ steps.check-authorized-user.outputs.authorized_user}}
+    environment: 'internal'
+    steps:
+      - name: Check identity
+        id: check-authorized-user
+        shell: bash
+        run: |
+          # AUTHORIZED_USERS is a newline separated list of usernames
+          if echo "${{ vars.AUTHORIZED_USERS }}" | grep -F -x -q "${{ github.actor }}"; then
+            echo "authorized_user=True" >> "$GITHUB_OUTPUT"
+          else
+            echo "authorized_user=False" >> "$GITHUB_OUTPUT"
+          fi
+  
+  # Authorization passes without approval if
+  # - the event is not a pull request (eg. push to main)
+  # - pull request comes from another branch in the same repo
+  # - author is in our predefined list of authorized users
+  # If none of these conditions are met, the workflow requires 
+  # manual approval from a maintainer with write permissions to continue
+  authorize:
+    needs: [check-identity]
+    environment: ${{ 
+      ( github.event_name != 'pull_request_target' ||
+        github.event.pull_request.head.repo.full_name == github.repository || 
+        needs.check-identity.outputs.authorized_user == 'True' )
+      && 'internal' || 'external' }}
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "Auth OK"
+
+  # Send a warning and fail this job if the workflow file was changed.
+  # Rest of the workflow continues as normal but the job failure will grab author's attention.
+  no-workflow-changes:
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'pull_request_target' }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.ref }}
+          fetch-depth: 0
+
+      - name: Check if workflow is modified
+        id: build-yml-changed
+        uses: tj-actions/changed-files@v40
+        with:
+          files: .github/workflows/test-ghaf-infra.yml
+
+      - name: Send warning
+        run: |
+          if [ "${{ steps.build-yml-changed.outputs.any_changed }}" == "true" ]; then
+            echo "::error::"\
+                 "This change edits workflow file '.github/workflows/test-ghaf-infra.yml'."\
+                 "Raising this error to notify that the build workflow change will only take impact after merge."\
```

Comment:
Same thing here too, consider removing the reference to 'build' workflow in the comment, for example to:

"Raising this error to notify that the workflow change will only take impact after merge."